lyncs_quda/dirac.py

```diff
@@ -7,38 +7,96 @@
 ]
 
 from functools import wraps
+from dataclasses import dataclass, field, InitVar
+from numpy import sqrt
 from dataclasses import dataclass
 from lyncs_cppyy import make_shared, nullptr
 from .gauge_field import gauge, GaugeField
+from .clover_field import CloverField
 from .spinor_field import spinor
 from .lib import lib
 from .enums import QudaPrecision
 
 
 @dataclass
 class Dirac:
-    gauge: GaugeField
-    clover: GaugeField = None
-    clover_inv: GaugeField = None
-    precond: GaugeField = None
+
+    # For QUDA DiracParam class:
+    gauge: GaugeField  # TODO: check acceptable geometry of gauge as an argument to QUDA Dirac class
```

Comment:
make `gauge` a frozen attribute and don't worry about it changing, i.e. remove `_gauge` and property